ships/0020-shipwright-image-api.md

```diff
@@ -0,0 +1,123 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: shipwright-image-api
+authors:
+  - "@imjasonh"
+reviewers:
+  - "@shbose78"
+  - "@adambkaplan"
+  - "@SaschaSchwarze0"
+  - "@ricardomaraschini"
+approvers:
+  - "@adambkaplan"
+  - "@SaschaSchwarze0"
+creation-date: 2021-08-16
+last-updated: 2021-08-16
+status: provisional
+see-also:
+  - https://github.com/shipwright-io/community/issues/8
+---
+
+# Shipwright Image API
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions
+
+1. Is this a worthwhile and well-aligned addition to the existing Shipwright project scope?
+
+1. Should Shipwright Image subproject release at the same cadence as the Shipwright Build subproject? (don't need to answer soon)
+
+## Summary
+
+Add a subproject to the Shipwright project that focuses on describing a set of general-purpose `Image` APIs.
+This may include a registry-protocol proxy for in-cluster or out-of-cluster image registries, that updates `Image` resources when images change, and can use Kubernetes RBAC rules on `Image` resources to guard pushes/pulls/etc of proxied image resources.
```

Comment:
We certainly gonna circle back on this "registry-protocol proxy" later but just for sake of clarity: the current implementation does not yet speak "registry protocol" but it is just a GRPC service dealing with compressed image files.